modules/related-posts/jetpack-related-posts.php

```diff
@@ -248,6 +253,127 @@ public function get_target_html_unsupported() {
 		return "\n\n<!-- Jetpack Related Posts is not supported in this context. -->\n\n";
 	}
 
+	/**
+	 * ===============
+	 * GUTENBERG BLOCK
+	 * ===============
+	 */
+
+	/**
+	 * Render the related posts markup.
+	 *
+	 * @param array $attributes Block attributes.
+	 * @return string
+	 */
+	public function render_block( $attributes ) {
+		$block_attributes = array(
+			'show_thumbnails' => isset( $attributes['displayThumbnails'] ) && $attributes['displayThumbnails'],
+			'show_date'       => isset( $attributes['displayDate'] ) ? (bool) $attributes['displayDate'] : true,
+			'show_context'    => isset( $attributes['displayContext'] ) && $attributes['displayContext'],
+			'layout'          => isset( $attributes['postLayout'] ) && 'list' === $attributes['postLayout'] ? $attributes['postLayout'] : 'grid',
+			'size'            => ! empty( $attributes['postsToShow'] ) ? absint( $attributes['postsToShow'] ) : 3,
+		);
+
+		$excludes = array();
+		if ( isset( $_GET['relatedposts_origin'] ) ) {
+			if ( is_string( $_GET['relatedposts_origin'] ) ) {
+				$excludes = explode( ',', $_GET['relatedposts_origin'] );
+			} elseif ( is_array( $_GET['relatedposts_origin'] ) ) {
+				$excludes = array_values( $_GET['relatedposts_origin'] );
+			}
+
+			$excludes = array_unique( array_filter( array_map( 'absint', $excludes ) ) );
+		}
+
+		$related_posts = $this->get_for_post_id(
+			get_the_ID(),
+			array(
+				'size'             => $block_attributes['size'],
+				'exclude_post_ids' => $excludes,
+			)
+		);
+
+		if ( ! $related_posts ) {
+			return '';
+		}
+
+		remove_filter( 'the_content', array( $this, 'filter_add_target_to_dom' ), 40 );
+
+		ob_start();
+		?>
+		<div id="jp-relatedposts" class="jp-relatedposts jp-relatedposts-block" style="display: block;">
```

Comment:
This is another area that I wish we could pull out to avoid code duplication, although I realize it would require a lot of changes in the way the existing Related Posts work.

Reply:
Yeah, we'd need to change a bunch, and I preferred not to, as it was going way beyond I'd expect to do for building a block. I realize we introduce some duplication here, but this way we're getting the best of both worlds - reusing as much as possible from the other functionality, while not letting unwanted functionality to stand in our way, but without refactoring a bunch of the existing module.